src/checkout/checkout_service.bal

```diff
@@ -0,0 +1,182 @@
+import ballerina/grpc;
+import ballerina/uuid;
+import ballerina/log;
+
+listener grpc:Listener ep = new (9094);
+
+configurable string cartUrl = "localhost";
+configurable string catalogUrl = "localhost";
+configurable string currencyUrl = "localhost";
+configurable string shippingUrl = "localhost";
+configurable string paymentUrl = "localhost";
+configurable string emailUrl = "localhost";
+
+@grpc:ServiceDescriptor {descriptor: ROOT_DESCRIPTOR_DEMO, descMap: getDescriptorMapDemo()}
+service "CheckoutService" on ep {
+    final CartServiceClient cartClient;
+    final ProductCatalogServiceClient catalogClient;
+    final CurrencyServiceClient currencyClient;
+    final ShippingServiceClient shippingClient;
+    final PaymentServiceClient paymentClient;
+    final EmailServiceClient emailClient;
+
+    function init() returns error? {
+        self.cartClient = check new ("http://" + cartUrl + ":9092");
+        self.catalogClient = check new ("http://" + catalogUrl + ":9091");
+        self.currencyClient = check new ("http://" + currencyUrl + ":9093");
+        self.shippingClient = check new ("http://" + shippingUrl + ":9095");
+        self.paymentClient = check new ("http://" + paymentUrl + ":9096");
+        self.emailClient = check new ("http://" + emailUrl + ":9097");
+    }
+
+    remote function PlaceOrder(PlaceOrderRequest value) returns PlaceOrderResponse|error {
+        string orderId = uuid:createType1AsString();
+        //Get Cart of the user
```

Comment:
These comments are unnecessary since it is intuitive to get the idea from the code itself.